core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -378,11 +375,13 @@ private[log] class Cleaner(val id: Int,
    * @param segments The group of segments being cleaned
    * @param map The offset map to use for cleaning segments
    * @param deleteHorizonMs The time to retain delete tombstones
+   * @param stats Collector for cleaning statistics
    */
   private[log] def cleanSegments(log: Log,
                                  segments: Seq[LogSegment], 
                                  map: OffsetMap, 
-                                 deleteHorizonMs: Long) {
+                                 deleteHorizonMs: Long,
+                                 stats: CleanerStats = new CleanerStats()) {
```

Comment:
Do we want the default? Generally, it's better to avoid defaults like this as it can lead to bugs where one forgets to pass the `stats` instance. Same for other similar cases.